src/components/ItemList/ItemList.tsx

```diff
@@ -13,31 +14,57 @@ export type Props<T extends { id: string }> = {
 
 const ItemList = <T extends { id: string }>({
     activeItemIndex = 0,
+    autoScroll = false,
     itemRowAs: ItemRowComponent = ItemRow,
     items,
     onActivate = noop,
     onSelect,
     ...rest
-}: Props<T>): JSX.Element => (
-    <ul className="ba-ItemList" data-testid="ba-ItemList" role="listbox" {...rest}>
-        {items.map((item, index) => (
-            <ItemRowComponent
-                key={item.id}
-                className="ba-ItemList-row"
-                isActive={index === activeItemIndex}
-                item={item}
-                onClick={(event: React.SyntheticEvent) => {
-                    onSelect(index, event);
-                }}
-                onMouseDown={(event: React.SyntheticEvent) => {
-                    event.preventDefault(); // preventDefault on mousedown so blur doesn't happen before click
-                }}
-                onMouseEnter={() => {
-                    onActivate(index);
-                }}
-            />
-        ))}
-    </ul>
-);
+}: Props<T>): JSX.Element => {
+    const listRef = React.useRef<HTMLUListElement>(null);
+
+    const activeRef = React.useCallback(
+        (activeEl: ItemRowRef) => {
+            const { current: listEl } = listRef;
+
+            if (!autoScroll || !activeEl || !listEl) {
+                return;
+            }
+
+            const { offsetHeight: activeHeight, offsetTop: activeTop } = activeEl;
+            const { offsetHeight: listHeight, scrollTop: listTop } = listEl;
+
+            if (listTop > activeTop) {
```

Comment:
Tests?

Reply:
The `offsetTop` and `offsetHeight` etc. are always 0 in jsdom. I think there is no way to test `useCallback` now. I will keep this in mind and test it in e2e tests.

Reply:
High unit test coverage is an important goal for the project, so we should at least attempt to test all component logic. You should be able spy on both `React.useRef` and `React.useCallback` and pass them DOM elements with those properties defined/mocked ([example](https://github.com/box/box-annotations/blob/master/src/image/__tests__/ImageAnnotator-test.ts#L34)).